Modules/_io/textio.c

```diff
@@ -248,19 +248,19 @@ _io_IncrementalNewlineDecoder___init___impl(nldecoder_object *self,
                                             PyObject *errors)
 /*[clinic end generated code: output=fbd04d443e764ec2 input=89db6b19c6b126bf]*/
 {
-    self->decoder = decoder;
-    Py_INCREF(decoder);
-
     if (errors == NULL) {
-        self->errors = _PyUnicode_FromId(&PyId_strict);
+        Py_XSETREF(self->errors, _PyUnicode_FromId(&PyId_strict));
         if (self->errors == NULL)
             return -1;
     }
     else {
-        self->errors = errors;
+        Py_XSETREF(self->errors, errors);
     }
     Py_INCREF(self->errors);
```

Comment:
I think it is simpler:
```suggestion
    if (errors == NULL) {
        errors = _PyUnicode_FromId(&PyId_strict);
        if (errors == NULL)
            return -1;
    }
    Py_INCREF(errors);
    Py_XSETREF(self->errors, errors);```